File: src/task/services/health.mjs
import { healthProcessor } from "#/task/processors/index.mjs";

async function ping(call, callback) {
  try {
    if (call.request.throw) throw Error("This is a test error.");
    const response = await healthProcessor.ping();
    callback(null, response);
  } catch (e) {
    callback(e, null);
  }
}

export default { ping };

File: src/tracker/services/health.py
from tracker.processors import HealthProcessor
from tracker.protobuf.v1 import health_pb2, health_pb2_grpc


class HealthService(health_pb2_grpc.HealthServiceServicer):
    """Servicer: Health Check Service"""

    processor = HealthProcessor()

    async def Ping(self, request, context):
        """Ping SQL"""
        if request.throw:
            msg = "This is a test error."
            raise ValueError(msg)
        reply: str = await self.processor.ping()  # type: ignore[annotation-unchecked]
        return health_pb2.PingResponse(reply=reply)
